editor/plugins/version_control_editor_plugin.cpp

```diff
@@ -232,6 +233,20 @@ void VersionControlEditorPlugin::_refresh_branch_list() {
 	}
 }
 
+String VersionControlEditorPlugin::_get_date_string_from(int64_t p_unix_timestamp, int64_t p_offset_minutes) const {
+	char sign = p_offset_minutes > 0 ? '+' : '-';
+	int unsigned_offset = Math::abs((int)p_offset_minutes);
+	int hours = unsigned_offset / 60;
+	int minutes = unsigned_offset % 60;
+
+	String offset_string = String::chr(sign);
+	offset_string += (hours < 10 ? "0" : "") + String::num_int64(hours);
+	offset_string += ":";
+	offset_string += (minutes < 10 ? "0" : "") + String::num_int64(minutes);
+
+	return vformat("%s %s", Time::get_singleton()->get_datetime_string_from_unix_time(p_unix_timestamp + p_offset_minutes * 60, true), offset_string);
+}
```

Comment:
@aaronfranke Would this be an intended use of the Time singleton?

Reply:
Yes. To keep it simple, the Time singleton does not deal with timezone offsets or suffixes, so including them manually is the way to do it (assuming you do in fact need the offset).

However, note that it's convention to not put a space between the timestamp and offset (so it would be `"%s%s"`), and for zero offset it's common to abbreviate this to `Z` (so if `p_offset_minutes` is zero, we can optionally do a version with `"%sZ"`)

Another consideration, we could potentially shorten this using `get_time_string_from_unix_time`, or maybe there should be a dedicated method for converting an offset number to a string - even if the rest of the methods are unaware of timezones, we can add a method to do the same logic you're doing here for the suffix. We just need to think about things like if we want it to print `Z` or `+00:00` and if we want the input value to be in minutes or seconds.

Reply:
I opened #58485 for this.

Reply:
Thanks for adding the offset to string conversion to `Time`, I will rebase this PR once that change is merged. 

The reason for me not going for a `%sZ%s` is that Git usually shows the `+05:30` format rather than the one with a `Z`, so I will probably just change it to directly use the offset to string conversion function. 

Since the VCS integration is closely (but not directly) mimicing Git, for the time being I am trying to incorporate "Git-likeness" in it if it seems fine, and that is up until the time where we work on other VCS plugins that would require us to further generalise the time format used in the VCS integration in the future, amongst tons of other generalizations that we'd need to make in the UI.